library/std/src/net/ip.rs

```diff
@@ -1177,21 +1401,27 @@ impl Ipv6Addr {
     /// # Examples
     ///
     /// ```
-    /// #![feature(ip)]
-    ///
     /// use std::net::Ipv6Addr;
     ///
     /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0xffff, 0xc00a, 0x2ff).is_global(), true);
     /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 0x1).is_global(), false);
     /// assert_eq!(Ipv6Addr::new(0, 0, 0x1c9, 0, 0, 0xafc8, 0, 0x1).is_global(), true);
     /// ```
+    ///
+    /// # Stability guarantees
+    ///
+    /// This method's behavior [may by subject to changes][changes] in the
+    /// future, as new IETF RFCs are published.
+    ///
+    /// [changes]: super#stability-guarantees-for-ietf-defined-behavior
+    #[stable(feature = "ip", since = "1.48.0")]
     #[rustc_const_unstable(feature = "const_ipv6", issue = "76205")]
     pub const fn is_global(&self) -> bool {
-        match self.multicast_scope() {
+        (match self.multicast_scope() {
             Some(Ipv6MulticastScope::Global) => true,
             None => self.is_unicast_global(),
             _ => false,
-        }
+        }) || if let Some(v4_addr) = self.to_ipv4() { v4_addr.is_global() } else { false }
     }
```

Comment:
Two comments regarding the [IANA IPv6 special purpose address registry](https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml):

Technically all IPv4-mapped addresses are designated as not globally reachable. However, the entire point of those addresses is to represent IPv4 addresses. So while e.g. `::ffff:1.1.1.1` is technically not globally reachable, most? applications will treat it as `1.1.1.1`, which is globally reachable. What should Rust do here? 

Also, this implementation ignores some other addresses in the registry designated as not globally reachable, such as IPv6 benchmarking addresses: `2001:2::/48`.

Reply:
> Technically all IPv4-mapped addresses are designated as not globally reachable. However, the entire point of those addresses is to represent IPv4 addresses. So while e.g. `::ffff:1.1.1.1` is technically not globally reachable, most? applications will treat it as `1.1.1.1`, which is globally reachable. What should Rust do here?

I really don't know one way or the other which option would be "more correct." My gut feeling is that it makes sense to kick the can down the road, and ignore IPv4-mapped addresses in this context, the reason being that if someone is trying to make their IP globally routable they likely won't want it to be IPv4-mapped.

> Also, this implementation ignores some other addresses in the registry designated as not globally reachable, such as IPv6 benchmarking addresses: 2001:2::/48.

What do you think should be done to address this?

Reply:
If we specifically don't want to forward this one to mapped V4 addresses we could call that out in the documentation with an example that performs the conversion for anybody that does want it.

Otherwise, is it defensible to make these methods all behave consistently with respect to mapped V4 addresses and always forward to them?